software_training_assignment/src/moveTurtle.cpp

```diff
@@ -4,43 +4,80 @@ Action server for moving the turtle
 
 #include <ros/ros.h> 
 #include <actionlib/server/simple_action_server.h>
+#include <software_training_assignment/moveTurtle.h>
 //Subscribe to the geometry_msgs/Twist topic 
 #include <geometry_msgs/Twist.h>
+#include <math.h>
 
-bool moveTurtle(int x, int y){
-    
+//Class to define the action server
+class MoveTurtleAction {
+    protected: 
+        ros::NodeHandle nh; 
+        actionlib::SimpleActionServer<software_training_assignment::moveTurtle> as;
+        std::string action_name; 
+        software_training_assignment::moveTurtleFeedback feedback; 
+        software_training_assignment::moveTurtleResult result; 
 
-    return success; 
-}
+    public: 
 
-int main(int argc, char **argv) {
-    ros::init(argc, argv, "moveTurtle"); 
-    ros::NodeHandle nh; 
+        MoveTurtleAction(std::string name) : 
```

Comment:
assign nodehandle here

Reply:
Do you mean to put nodehandle in its own class? 
Currently nodehandle is in the MoveTurtleActionClass: 
`class MoveTurtleAction {
    protected: 
        ros::NodeHandle nh; `